zonec.c

```diff
@@ -1215,6 +1599,116 @@ zadd_rdata_txt_clean_wireformat()
 	}
 }
 
+static int
+svcparam_key_cmp(const void *a, const void *b)
+{
+	return ((int)read_uint16(rdata_atom_data(*(rdata_atom_type *)a)))
+	     - ((int)read_uint16(rdata_atom_data(*(rdata_atom_type *)b)));
+}
+
+void
+zadd_rdata_svcb_check_wireformat(uint16_t rr_type)
+{
+	size_t i, j;
+	uint8_t paramkeys[SVCPARAMKEY_COUNT - 1];
```

Comment:
-1 Stack violation with ipv6hints, which probably did not result in crash because of aligning on 32bits or 64bits.
```suggestion
	uint8_t paramkeys[65536];
```